napari/layers/_shapes_layer/model.py

```diff
@@ -444,22 +499,30 @@ def block_update_properties(self):
     def _get_shape(self):
         """Determines the shape of the vertex data.
         """
-        if len(self.data._vertices) == 0:
-            return [1, 1]
+        if len(self._data_view._vertices) == 0:
+            slice_shape = [1, 1]
+        else:
+            slice_shape = list(np.max(self._data_view._vertices, axis=0) + 1)
+
+        if len(self.data.keys()) == 1:
+            return [1] * (self._input_ndim-2) + slice_shape
         else:
-            return np.max(self.data._vertices, axis=0) + 1
+            keys = list(self.data.keys())
+            keys.remove(())
+            max_val = np.array(keys).max(axis=0)
+            return list(max_val) + slice_shape
```

Comment:
As a minor comment, it kind of makes me tick a little when a shape is a list and not a tuple. I think everything you've done here can be done with tuples, e.g. `(1,) * (self._input_ndim - 2) + slice_shape`, so I would prefer that. This is very minor though and can be left till later.

Reply:
changed to tuple